src/identity/apis/auth.test.ts

```diff
@@ -0,0 +1,281 @@
+import {mocked} from 'ts-jest/utils'
+
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+import {
+  getIdentity,
+  Error as UnityError,
+  Identity,
+} from 'src/client/unityRoutes'
+
+import {
+  retryFetchIdentity,
+  UnauthorizedError,
+  ServerError,
+} from 'src/identity/apis/auth'
+
+jest.mock('src/shared/constants', () => ({CLOUD: true}))
+
+jest.mock('src/shared/utils/featureFlag.ts', () => ({
+  isFlagEnabled: jest.fn(),
+}))
+
+jest.mock(
+  'src/client/unityRoutes',
+  () => ({
+    ...jest.requireActual('src/client/unityRoutes'),
+    getIdentity: jest.fn(),
+  }),
+  {virtual: true}
+)
+
+const mockIdentity: Identity = {
+  user: {
+    id: '1234asdf',
+    email: 'test@example.com',
+    accountCount: 1,
+    orgCount: 1,
+  },
+  account: {
+    id: 12344,
+    name: 'John Doe',
+    type: 'free',
+    accountCreatedAt: 'anytime',
+  },
+  org: {
+    id: '7890uiop',
+    name: 'John Doe org',
+    clusterHost: 'aws-east-0-1',
+  },
+}
+
+const mockGoodResponse = () => {
+  return Promise.resolve({
+    status: 200,
+    headers: {},
+    data: mockIdentity,
+  } as any)
+}
+
+const mockError401: UnityError = {
+  code: 'unauthorized',
+  message: "You ain't allowed to see that, jack",
+}
+
+const mockUnauthorizedError = () => {
+  return Promise.resolve({
+    status: 401,
+    headers: {},
+    data: mockError401,
+  } as any)
+}
+
+const mockUnityServerError: UnityError = {
+  code: 'internal error',
+  message: 'There was an error',
+}
+
+const mockServerError = () => {
+  return Promise.resolve({
+    status: 500,
+    headers: {},
+    data: mockUnityServerError,
+  } as any)
+}
+
+const flushPromises = () => {
+  return new Promise(jest.requireActual('timers').setImmediate)
+}
+
+describe('retrying failed authentication attempts', () => {
+  beforeEach(() => {
+    // return true so that `uiUnificationFlag` and `quartzIdentity` both return true
```

Comment:
@wdoconnell it's more that since the call is being mocked in this test, it doesn't really matter if it's making the call to `/me` or `/quartz/me` or `/quartz/identity`. The logic being tested is how the response is handled.

Although now that I think about it `/quartz/me` can return a 404, which isn't handled here 🤔